app/models/mixins.py

```diff
@@ -0,0 +1,31 @@
+import base64
+from sqlalchemy.dialects.postgresql import BIGINT
+
+from app import db
+from app.utils import random_positive_public_id
+
+
+class PublicIdMixin:
+    public_id = db.Column(BIGINT, default=random_positive_public_id, nullable=False, index=True, unique=True)
+
+    @staticmethod
+    def __pad_as_bytes(val):
+        """Converts an int to its equivalent in bytes and prepends with \x00 bytes until it's a multiple of 8 bytes
+        long to create ciphertexts of the same length."""
+        return val.to_bytes(8, 'big')
+
+    @staticmethod
+    def __append_equals(text):
+        """Appends '=' to the given text until it's a multiple of 8 characters long and so suitable for base32
+        decoding."""
+        return text + ''.join('=' * (4 - (len(text) % 4)))
```

Comment:
I don't really get why we need this - if someone passes something that isn't 8 characters long, then surely it isn't one of our identifiers, so we shouldn't be trying to decode it?

Reply:
Our encoding function strips any extra '=' on the end, because I think it looks ... nicer and it's then just an alphanumeric string, which I expect users would feel more comfortable with. It's arguable whether we want to do this or not, but if we do, then we need to put them back on when the IDs come back.

Reply:
Ah right yeah I see what you mean, they are ugly.

On the assumption that they're always 16 characters long I'd be tempted to write this as it's more clear? `text.ljust(16, '=')`